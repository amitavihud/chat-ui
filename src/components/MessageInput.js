import React from 'react'
import styled, { css } from "styled-components";

const Container = styled.div`
  padding: 10px;
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: auto auto auto;
  
  .avatar {
    width: 20px;
    align-self: center;
    justify-self: center;
  }
`
const inputsStyle = css`
  background-color: #F1F1F7;
  color: #858DA8;
  border: none;
  border-radius: 3px;
  outline-color: transparent;
  outline-style: none;
  width: 100%;
  box-sizing: border-box;
  &::placeholder {
    color: #c4cadc
  }
`

const Input = styled.input`
  ${inputsStyle}
  padding: 2px 9px;
`
const Message = styled.textarea`
  ${inputsStyle}
  padding: 10px;
  margin-top: 5px;
  resize: none;
`
const NameInputContainer = styled.div`
  display: grid;
  grid-template-columns: 30px auto;
`
const SendContainer = styled.div`
  justify-self: right;
`
const Send = styled.button`
  padding: 8px 15px;
  margin: 10px;
  background-color: #50BC7E;
  color: white;
  border-radius: 5px;
  outline-style: none;

  &:active {
    background-color: #289154;
  }
`

export default class MessageInput extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      name: '',
      message: ''
    }

    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleNameChange = this.handleNameChange.bind(this)
    this.handleMessageChange = this.handleMessageChange.bind(this)
  }
  static getDerivedStateFromProps(props, state){
    if (!state.name) {
      return {
        name: props.username
      }
    }
    return null
  }
  handleSubmit() {
    this.props.onSubmit({
      name: this.state.name,
      message: this.state.message,
      avatarURL:this.props.avatarURL
    })
    this.setState({
      message: ''
    })
  }
  handleNameChange(e) {
    this.setState({
      name: e.target.value
    })
  }
  handleMessageChange(e) {
    this.setState({
      message: e.target.value
    })
  }
  render() {
    return (
      <Container>
        <NameInputContainer>
          <img className="avatar" src={this.props.avatarURL} />
          <Input placeholder="Your name" onChange={this.handleNameChange} value={this.state.name} />
        </NameInputContainer>
        <Message placeholder="Message" onChange={this.handleMessageChange} value={this.state.message} />
        <SendContainer>
          <Send onClick={this.handleSubmit}>Send</Send>
        </SendContainer>        
      </Container>
    )
  }
}