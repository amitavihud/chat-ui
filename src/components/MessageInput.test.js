import React from 'react'
import renderer from 'react-test-renderer';
import MessageInput from './MessageInput'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

describe('MessageInput component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MessageInput />
    )

    expect(tree).toMatchSnapshot()
  })

  describe('the message field is empty', () => {
    it('does not allow the user to send her message', () => {
      const wrapper = mount(<MessageInput />)
      const send = wrapper.find('button')

      expect(send.is('[disabled]')).toBe(true)
    })
  })

  describe('the message field contains something', () => {
    it('allows the user to send her message', () => {
      const wrapper = mount(<MessageInput />)

      wrapper.find('textarea').value = 'something'
      wrapper.find('textarea').simulate('change')
      
      const send = wrapper.find('button')
      expect(send.is('[disabled]')).toBe(true)
    })
  })
})