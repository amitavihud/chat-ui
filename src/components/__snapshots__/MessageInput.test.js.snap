// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput component renders correctly 1`] = `
<div
  className="sc-bdVaJa kgIdzV"
>
  <div
    className="sc-bxivhb ieUFMR"
  >
    <img
      className="avatar"
      src={undefined}
    />
    <input
      className="sc-bwzfXH iqTZal"
      onChange={[Function]}
      placeholder="Your name"
      value={undefined}
    />
  </div>
  <textarea
    className="sc-htpNat guXHJY"
    onChange={[Function]}
    placeholder="Message"
    value=""
  />
  <div
    className="sc-ifAKCX dmLNXp"
  >
    <button
      className="sc-EHOje gHrGfF"
      disabled={true}
      onClick={undefined}
    >
      Send
    </button>
  </div>
</div>
`;
